scipy/integrate/_ivp/ivp.py

```diff
@@ -18,12 +19,53 @@
 
 
 MESSAGES = {0: "The solver successfully reached the end of the integration interval.",
-            1: "A termination event occurred."}
+            1: "A termination event occurred.",
+            2: "Solver stopped via callback."}
 
 
 class OdeResult(OptimizeResult):
     pass
 
+class IntermediateOdeResult(_RichResult):
+    """
+    Represents an in-progress ode result.
+
+    Attributes
+    ----------
+    t : list, length (n_points)
+        Time points of solution up to and including current time step.
+        Unlike the result from `solve_ivp`, this is a list.
+    y : ndarray, shape (n, n_points)
```

Comment:
Is this a list?

Reply:
Yes.  iIf we keep this attribute with history, this description will need to be changed.  It is something like `list [ list | ndarray]`, but it needs investigating to get all the edge cases correct.  I will hold off on doing that until we agree on whether it should be included, unless it affects the decision. It is also transposed with respect to the final return of `solve_ivp`.